tools/clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -15119,16 +15177,32 @@ static bool nodeInputIsCompatible(StringRef &typeName,
       .Default(false);
 }
 
-void DiagnoseNodeEntry(Sema &S, FunctionDecl *FD, HLSLShaderAttr *Attr) {
+void DiagnoseComputeEntry(Sema &S, FunctionDecl *FD, llvm::StringRef StageName,
+                          bool isActiveEntry) {
+  if (isActiveEntry) {
+    if (auto WaveSizeAttr = FD->getAttr<HLSLWaveSizeAttr>()) {
+      std::string profile = S.getLangOpts().HLSLProfile;
+      const ShaderModel *SM = hlsl::ShaderModel::GetByName(profile.c_str());
+      if (!SM->IsSM66Plus()) {
+        S.Diags.Report(WaveSizeAttr->getRange().getBegin(),
+                       diag::err_hlsl_attribute_in_wrong_shader_model)
+            << "wavesize"
+            << "6.6";
```

Comment:
While this is fine for now, I think as these kinds of checks expand to more attributes, we can share more code, and shouldn't be doing things like translating the profile string to a shader model locally.

In other words, we would want to:
- Pass ShaderModel down to `Diagnose*Entry` functions when they need it
- Create a generic function to wrap a shader model version check for an attribute

Once that's done, the `DiagnoseNodeEntry`, or `DiagnoseComputeEntry` code for this would just look like:
```hlsl
if (auto WaveSizeAttr = FD->getAttr<HLSLWaveSizeAttr>()) {
  if (isActiveEntry)
    DiagnoseAttrRequiresShaderModel(WaveSizeAttr, 6, 6, SM);
  // do other WaveSize checks here...
  // such as ensuring WaveSize divides evenly into NumThreads when WaveMatrix is used.
}
```